metatron/src/test/java/uk/gov/ida/eidas/metatron/apprule/rules/MetatronTests.java

```diff
@@ -136,13 +173,26 @@ private Client getClient() {
         return client;
     }
 
+    private void assertThatCertsMatch(String samlEncryptionCertX509, String testRpPublicEncryptionCert) {
+        assertThat(StringUtils.right(samlEncryptionCertX509, 10)).isEqualTo(StringUtils.right(testRpPublicEncryptionCert, 10));
```

Comment:
Is it not possible to compare the entire string?  These sort of comparison tests always make me nervous because we're never sure they're truly accurate.

Would you consider turning both the strings into certs and doing `.isEqualToComparingFieldByField()` if the whole strings can't be compared for some reason?

Reply:
👍, tests now compare X509Certificates for equality